crates/sargon/src/system/sargon_os/transactions/support/signable_manifest_summary.rs

```diff
@@ -0,0 +1,151 @@
+use crate::prelude::*;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct SignableManifestSummary {
+    pub id: Exactly32Bytes,
+    pub summary: ManifestSummary,
+}
+
+impl SignableManifestSummary {
+    pub fn new(summary: ManifestSummary) -> Self {
+        Self {
+            id: Exactly32Bytes::generate(),
+            summary: summary,
+        }
+    }
+}
+
+impl HasSampleValues for SignableManifestSummary {
+    fn sample() -> Self {
+        Self {
+            id: Exactly32Bytes::sample(),
+            summary: ManifestSummary::sample(),
+        }
+    }
+
+    fn sample_other() -> Self {
+        Self {
+            id: Exactly32Bytes::sample_other(),
+            summary: ManifestSummary::sample_other(),
+        }
+    }
+}
+
+impl SignableID for Exactly32Bytes {}
+
+impl Into<Exactly32Bytes> for SignableManifestSummary {
+    fn into(self) -> Exactly32Bytes {
+        self.id.clone()
+    }
+}
+
+impl std::hash::Hash for SignableManifestSummary {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.id.hash(state);
+    }
+}
+
+impl Signable for SignableManifestSummary {
+    type ID = Exactly32Bytes;
+    type Payload = Self;
+
+    fn entities_requiring_signing(
+        &self,
+        profile: &Profile,
+    ) -> Result<IndexSet<AccountOrPersona>> {
+        ExtractorOfEntitiesRequiringAuth::extract(profile, self.summary.clone())
+    }
+
+    fn get_id(&self) -> Self::ID {
+        self.id.clone()
+    }
+
+    fn get_payload(&self) -> Self::Payload {
+        panic!("Manifest summary cannot be actually signed")
+    }
+
+    fn sample_entity_addresses_with_pub_key_hashes_requiring_auth(
```

Comment:
Lets refactor this.

Create a function on `TransactionManifest` which 

```rust

    fn sample_entity_addresses_with_pub_key_hashes_requiring_auth(
        account_addresses_requiring_auth: impl IntoIterator<
            Item = (AccountAddress, PublicKeyHash),
        >,
        identity_addresses_requiring_auth: impl IntoIterator<
            Item = (IdentityAddress, PublicKeyHash),
        >,
    ) -> Self {
        let mut network_id: Option<NetworkID> = None;

        let all_addresses_with_hashes = account_addresses_requiring_auth
            .into_iter()
            .map(|(address, hash)| {
                (AddressOfAccountOrPersona::from(address), hash)
            })
            .chain(identity_addresses_requiring_auth.into_iter().map(
                |(address, hash)| {
                    (AddressOfAccountOrPersona::from(address), hash)
                },
            ))
            .collect::<Vec<_>>();

        all_addresses_with_hashes
            .iter()
            .for_each(|(address, _hash)| {
                if let Some(network_id) = network_id {
                    assert_eq!(network_id, address.network_id())
                } else {
                    network_id = Some(address.network_id())
                }
            });

        let mut builder = ScryptoTransactionManifestBuilder::new();
        let network_id = network_id.unwrap_or_default();

        for (address, hash) in all_addresses_with_hashes {
            builder = builder.set_metadata(
                address.scrypto(),
                MetadataKey::OwnerKeys,
                ScryptoMetadataValue::PublicKeyHashArray(vec![hash.into()]),
            );
        }

        Self::sargon_built(builder, network_id)
    }

```

then refactor the function https://github.com/radixdlt/sargon/blob/ab1ae0c61d61b750680eb9c884b6ea975fd9ad30/crates/sargon/src/signing/signables/signable_transaction_intent.rs#L17 to use this, so that becomes:

```rust
   fn sample_entity_addresses_with_pub_key_hashes_requiring_auth(
        account_addresses_requiring_auth: impl IntoIterator<
            Item = (AccountAddress, PublicKeyHash),
        >,
        identity_addresses_requiring_auth: impl IntoIterator<
            Item = (IdentityAddress, PublicKeyHash),
        >,
    ) -> Self {
        let manifest = TransactionManifest::sample_entity_addresses_with_pub_key_hashes_requiring_auth(account_addresses_requiring_auth, identity_addresses_requiring_auth);

        Self::new(TransactionHeader::sample(), manifest, Message::None).unwrap()
    }
```

and then **here** we can just:

```rust
   fn sample_entity_addresses_with_pub_key_hashes_requiring_auth(
        account_addresses_requiring_auth: impl IntoIterator<
            Item = (AccountAddress, PublicKeyHash),
        >,
        identity_addresses_requiring_auth: impl IntoIterator<
            Item = (IdentityAddress, PublicKeyHash),
        >,
    ) -> Self {
        let manifest = TransactionManifest::sample_entity_addresses_with_pub_key_hashes_requiring_auth(account_addresses_requiring_auth, identity_addresses_requiring_auth);

        Self::new(manifest)
    }
```

Reply:
Each signable has a different builder, but probably the public_key_hashes could be computed in the Signable trait

Reply:
Check latest state -> https://github.com/radixdlt/sargon/blob/833df2780983a2894ab94b748d41b7c22953119e/crates/sargon/src/system/sargon_os/transactions/support/signable_manifest_summary.rs#L67